spec/services/work_preservation_service_spec.rb

```diff
@@ -2,44 +2,22 @@
 require "rails_helper"
 
 RSpec.describe WorkPreservationService do
-  let(:draft_work) { FactoryBot.create :draft_work, doi: "10.34770/pe9w-x904" }
-
-  it "raises an exception if the work has not been approved" do
-    subject = described_class.new(work: draft_work)
-    expect { subject.preserve! }.to raise_error(StandardError, /Cannot preserve work/)
-  end
-
   describe "preserves to default location" do
     let(:approved_work) { FactoryBot.create :approved_work, doi: "10.34770/pe9w-x904" }
-    let(:prefix) { approved_work.s3_query_service.prefix }
-    let(:preservation_directory) { prefix + "princeton_data_commons/" }
+    let(:bucket_name) { approved_work.s3_query_service.bucket_name }
+    let(:path) { approved_work.s3_query_service.prefix }
+    let(:preservation_directory) { path + "princeton_data_commons/" }
 
     before do
       stub_request(:put, "https://example-bucket-post.s3.amazonaws.com/#{preservation_directory}").to_return(status: 200)
-      stub_request(:get, "https://example-bucket-post.s3.amazonaws.com/?list-type=2&max-keys=1000&prefix=#{prefix}").to_return(status: 200)
+      stub_request(:get, "https://example-bucket-post.s3.amazonaws.com/?list-type=2&max-keys=1000&prefix=#{path}").to_return(status: 200)
       stub_request(:put, "https://example-bucket-post.s3.amazonaws.com/#{preservation_directory}metadata.json").to_return(status: 200)
       stub_request(:put, "https://example-bucket-post.s3.amazonaws.com/#{preservation_directory}datacite.xml").to_return(status: 200)
     end
 
-    it "preserves a work to the default location" do
-      subject = described_class.new(work: approved_work)
+    it "preserves a work to the indicated location" do
+      subject = described_class.new(work_id: approved_work.id, bucket_name: bucket_name, path: path)
       expect(subject.preserve!).to eq "s3://example-bucket-post/#{preservation_directory}"
     end
   end
-
-  describe "preserve to custom location" do
```

Comment:
Just curious why this was removed?  Was it intentional?

Reply:
The service now always receives the location rather than default to something, so in a way it always uses a custom location. The test covers both cases, but I just realized the description might be misleading. I'll update it.

Reply:
Updated: https://github.com/pulibrary/pdc_describe/pull/1189/commits/b1f8f7735063846385f49162a89ce8912588a162